test/cleanup.go

```diff
@@ -80,5 +80,9 @@ func EnsureCleanup(t *testing.T, cleanup func()) {
 // EnsureTearDown will delete created names when the test ends, either via
 // t.Cleanup, or on interrupt via CleanupOnInterrupt.
 func EnsureTearDown(t *testing.T, clients *Clients, names *ResourceNames) {
+	if ServingFlags.DisableTestCleanup {
+		t.Log("Cleanup not enabled")
+		return
+	}
```

Comment:
I'd probably move this to the function that's being passed to `EnsureCleanup` 

Then the log message will be printed after the test completes

Reply:
Just a thought: Does it make sense to only skip cleanup when the test fails (vs. every time) ? I think this flag is mostly useful when you're running the whole test suite in which case there would be so many resources left at the end. If I'm running/debugging just a single test locally I can easily comment out the TearDown and don't need the flag. But maybe my understanding is wrong...

Reply:
@arturenault wdyt about what @mgencur thought?

Reply:
To move this forward let's only perform the cleanup when the test fails and update the name of the flag to reflect the new behaviour.

The way I see it - while debugging in our CI by using this flag I'll end up with a dump of the resource yaml along with the test output. Currently we don't have this information since all the resources are deleted whether the test succeeds or fails

Reply:
@dprotaso Do we keep the check on EnsureTearDown and change the flag to something like "disable-test-cleanup-on-fail"?